src/NUnitFramework/tests/Assertions/AssertFailTests.cs

```diff
@@ -104,13 +104,13 @@ public void AssertFailWorksWithMessageAndArgs()
             Assert.AreEqual(ResultState.Failure, result.ResultState);
             Assert.AreEqual("MESSAGE: 2+2=4", result.Message);
 
-            Assert.AreEqual(result.AssertionResults.Count, 1);
+            Assert.AreEqual(1, result.AssertionResults.Count);
```

Comment:
These inverted arguments are existing technical debt and not really part of this change, so I'd suggest fixing them in a separate PR.